### ISSUE TITLE
cli: flux create secret git returns no key found

### ISSUE THREAD

**Issue by marshallford** [fluxcd/flux2#1239]
### Describe the bug

Unable to create secret containing private key file. I suspect the error is coming from [this `golang.org/x/crypto/ssh` error](https://github.com/golang/crypto/blob/master/ssh/keys.go#L1123) in the function `ParseRawPrivateKey`, but I can't be sure.

### To Reproduce

```
mford@box ~/flux-private-key-file-demo
❯ ssh-keygen -t ed25519 -C "demo ed25519"
Generating public/private ed25519 key pair.
Enter file in which to save the key (/home/mford/.ssh/id_ed25519): ed25519
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Your identification has been saved in ed25519.
Your public key has been saved in ed25519.pub.
The key fingerprint is:
SHA256:Snw1W27LMi43HLWdK48XgGESwwH0ax53YHX+53NPdBo demo ed25519
The key's randomart image is:
+--[ED25519 256]--+
|     .oo+o  . .  |
|       .o.o. o   |
|        .o=o. .  |
|     .   +.*o  . |
|      o S o.++E.=|
|     . = o.+..+=o|
|      . ..o.o .++|
|        ..+o..o.+|
|         o...+. .|
+----[SHA256]-----+

mford@box ~/flux-private-key-file-demo 10s
❯ ssh-keygen -t rsa -b 4096 -C "demo rsa"
Generating public/private rsa key pair.
Enter file in which to save the key (/home/mford/.ssh/id_rsa): rsa
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Your identification has been saved in rsa.
Your public key has been saved in rsa.pub.
The key fingerprint is:
SHA256:NqowkpfBCFEz+mWdyvLEuCS7e+hHdbv7xRQrztJjrcY demo rsa
The key's randomart image is:
+---[RSA 4096]----+
|..+              |
| o o . .         |
|o   o o    .     |
|.+ * o .    o    |
|o B * . S. o     |
| = O   ++.=      |
|+.B . ..o* +     |
|.+.+ . .oE+      |
|++. .  .+o       |
+----[SHA256]-----+

mford@box ~/flux-private-key-file-demo
❯ ls
ed25519  ed25519.pub  rsa  rsa.pub

mford@box ~/flux-private-key-file-demo
❯ flux --version
flux version 0.12.0

mford@box ~/flux-private-key-file-demo
❯ flux create secret git example --url ssh://git@host:port/org/repo.git  --private-key-file rsa
✗ ssh: no key found

mford@box ~/flux-private-key-file-demo
❯ flux create secret git example --url ssh://git@host:port/org/repo.git --private-key-file ed25519
✗ ssh: no key found
```

### Expected behavior

Flux CLI command `flux create secret git` should support common? key formats.

### Additional context

- Kubernetes version: N/A
- Git provider: Generic
- Container registry provider: N/A

**Comment by hiddeco** [comment]
This has been fixed in `v0.12.1`, released a couple of seconds ago.

**Comment by marshallford** [comment]
The turnaround time on both the fix and the release is **amazing**. Thanks @hiddeco!
